pages/blog/_posts/2020-03-30-catch-imsi-from-vowifi-session.md

```diff
@@ -0,0 +1,157 @@
+as rarely as possible and a randomly-generated TMSI is sent instead.*
+
+* [IMSI article](https://en.wikipedia.org/wiki/International_mobile_subscriber_identity)
+
+### Security implications
+The IMSI is a secret identifier stored on the sim and can be exploited in many
+ways once known and bound to sim so changing UE will not help.
+
+Examples:
+* Locating user (UE)
+* Intercepting calls
+* Intercepting SMS (stealing two factor pin eg.)
+* ..and more
+
+## How VoWifi works
+When your phone is connected to datanetwork and with volte and vowifi enabled
+the device (UE) establish sip session directly to packetgateway via 4g, and via
+public internet to epdg (evolved packet data gateway) which is in essence a
+ipsec (ikev2) termination using SIM-AKA to authenticate UE. The ipsec comes into
+play since epdg is exposed publicly.
+
+I won't go further in depth for volte and vowifi since there are already
+excellent articles about the matter:
+* [Voice over WLAN](https://en.wikipedia.org/wiki/Voice_over_WLAN)
+* [Voice over LTE](https://en.wikipedia.org/wiki/Voice_over_LTE)
+
+![VoWifi topology](/img/blog/vowifi-imsi-leak/vowifi.jpg)
+
+## EPDG exposed on the public internet
+The Evolved Packet Data Gateway needs to be publicly available on the internet
+since UE needs to access it from an arbitrary no-trusted connection. The ipsec
+will secure and encrypt both the data and maintain the integrity of the
+connection throughout the session.
+
+The UE finds the epdg termination point by looking dns records partly following
+a convention decided by 3gpp and typically looks like this in DNS:
+
+```
+epdg.epc.mnc001.mcc242.pub.3gppnetwork.org. 3488 IN A 77.16.1.126
+epdg.epc.mnc001.mcc242.pub.3gppnetwork.org. 3488 IN A 77.16.1.125
+```
+
+From the DNS records we recognize the network operator and and the country code,
+in case corresponds to Norway (242) and Telenor (001).
+
+The DNS records are registered under under a delegated domain owned by GSMA and
+usually are redelegated to operator under their own umbrella like the example
+mnc001.mcc242.pub.3gppnetwork.org.
+
+## The problem
+When UE establish session to epdg it uses a vpn, a ipsec relationship using
+IKEv2 for authentication, encryption and integrity.
+
+So far the implementation works as intended and provides good security through
+encryption and security.
+
+The problem is not the VoWifi per see, by rather how ipsec establish the
+session. When UE connects to the epdg, it acts as a initiator and the epdg is
+inherently passive since it cannot know where from (ip) the UE will come from.
+
+![sim-aka flow](/img/blog/vowifi-imsi-leak/sim-aka.png)
+
+### EAP-AKA exposes identity
+Vowifi as mentioned earlier utilises an encryption protocol based on the widely
+adopted Extensible Authentication Protocol. EAP itself is just a protocol and
+does not define the contents of the data or how exact the data exchanges look
+like. EAP-AKA unfortunately exposes user identity in the clear unencrypted during
```

Comment:
> "EAP-AKA unfortunately exposes user identity in the clear unencrypted" 

sounds like it should be either:

> "EAP-AKA unfortunately exposes user identity as plaintext"

or

> "EAP-AKA unfortunately exposes the unencrypted user identity "

Reply:
Fixed